firebase-firestore/src/test/java/com/google/firebase/firestore/FilterTest.java

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class FilterTest {
+
+  @Test
+  public void equalTo() {
+    Filter filter = Filter.equalTo("x", "v");
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.equalTo(FieldPath.of("x"), "v"));
+    assertNotEquals(filter, Filter.equalTo("x", "z"));
+    assertNotEquals(filter, Filter.equalTo("y", "v"));
+    assertNotEquals(filter,Filter.notEqualTo("y", "v"));
```

Comment:
add assert where the fields match and the values match, but the operator is different.
```
assertNotEquals(filter,Filter.notEqualTo("x", "v"));
```

can be done for the other tests too

Reply:
Right. That was the intent. I'll fix.